File: web/src/types/bill.ts
/**
 * 账单相关类型定义
 * 账单用于管理定期重复的固定支出（如房租、水电费、订阅服务等）
 * 字段名与后端 JSON tag 完全对应
 */

/**
 * 重复规则枚举
 * - Daily: 每天重复
 * - Weekly: 每周重复
 * - Monthly: 每月重复
 * - Yearly: 每年重复
 */
export enum RepeatRule {
  Daily = 'daily',
  Weekly = 'weekly',
  Monthly = 'monthly',
  Yearly = 'yearly',
}

/**
 * 账单信息接口（对应后端 BillResp）
 */
export interface Bill {
  id: number                    // 账单唯一标识（后端 uint64）
  name: string                  // 账单名称
  amount: string                // 账单金额
  repeat_rule: RepeatRule       // 重复规则
  next_due: string              // 下次到期日
  source_id: number | null      // 支出账户ID
  category_id: number | null    // 分类ID
  notes: string                 // 备注
  created_at: string            // 创建时间
  updated_at: string            // 最后更新时间
}

/**
 * 创建账单请求接口（对应后端 CreateBillReq）
 */
export interface CreateBillReq {
  name: string                  // 账单名称（必填）
  amount: string                // 账单金额（必填，必须大于0）
  repeat_rule: RepeatRule       // 重复规则（必填）
  next_due: string              // 下次到期日期（必填，YYYY-MM-DD）
  source_id?: number | null     // 支出账户ID（可选）
  category_id?: number | null   // 分类ID（可选）
  notes?: string                // 备注（可选）
}

/**
 * 更新账单请求接口（对应后端 UpdateBillReq）
 * 所有字段均为可选，只传需要修改的字段
 */
export interface UpdateBillReq {
  name?: string                 // 账单名称
  amount?: string               // 账单金额
  repeat_rule?: RepeatRule      // 重复规则
  next_due?: string             // 下次到期日期
  source_id?: number | null     // 支出账户ID
  category_id?: number | null   // 分类ID
  notes?: string                // 备注
}

File: server/internal/model/bill.go
// Package model 定义了系统的数据模型，对应数据库表结构
package model

import (
	"time"

	"github.com/shopspring/decimal"
	"gorm.io/gorm"
)

// RepeatRule 重复规则枚举
type RepeatRule string

const (
	RepeatRuleDaily   RepeatRule = "daily"   // 每日
	RepeatRuleWeekly  RepeatRule = "weekly"  // 每周
	RepeatRuleMonthly RepeatRule = "monthly" // 每月
	RepeatRuleYearly  RepeatRule = "yearly"  // 每年
)

// Bill 账单模型，对应bills表
// 管理周期性账单，自动提醒到期付款
type Bill struct {
	ID         uint64          `gorm:"primaryKey;autoIncrement" json:"id"`
	UserID     uint64          `gorm:"not null;index" json:"user_id"`                // 所属用户ID
	Name       string          `gorm:"not null;size:100" json:"name"`                // 账单名称
	Amount     decimal.Decimal `gorm:"type:decimal(19,4);not null" json:"amount"`    // 账单金额
	RepeatRule RepeatRule      `gorm:"not null;size:20" json:"repeat_rule"`          // 重复规则（daily/weekly/monthly/yearly）
	NextDue    time.Time       `gorm:"not null" json:"next_due"`                     // 下次到期日期
	SourceID   *uint64         `json:"source_id"`                                    // 支出账户ID
	CategoryID *uint64         `json:"category_id"`                                  // 分类ID
	Notes      string          `gorm:"type:text" json:"notes"`                       // 备注信息
	CreatedAt  time.Time       `gorm:"not null" json:"created_at"`
	UpdatedAt  time.Time       `gorm:"not null" json:"updated_at"`
	DeletedAt  gorm.DeletedAt  `gorm:"index" json:"-"`                               // 软删除时间
}

// TableName 指定表名
func (Bill) TableName() string { return "bills" }
